java/src/jmri/util/FileUtilSupport.java

```diff
@@ -1396,7 +1440,7 @@ public void backup(File file) throws IOException {
      * @throws IllegalArgumentException if max is less than one
      * @see #backup(java.io.File)
      */
-    public void rotate(@Nonnull File file, int max, String extension) throws IOException {
+    public void rotate(@Nonnull File file, int max, @Nonnull String extension) throws IOException {
```

Comment:
extension is nullable, and should be annotated as such. Please see the Javadoc comments for this method concerning how a null extension is handled.